### ISSUE TITLE
GDBus support

### ISSUE THREAD

**Issue by ghost** [hexchat/hexchat#252]
Would be useful for many things:
- --existing support for Win32
- ~~--command support for Win32~~
- **working** one instance option on all platforms
- [x] dbus-client.c
- [ ] dbus-plugin.c

**Comment by DGMurdockIII** [comment]
here are some info you might want 
https://www.iana.org/assignments/uri-schemes/prov/irc
https://tools.ietf.org/html/draft-butcher-irc-url-04

**Comment by he7d3r** [comment]
I just noticed that, on Debian, hexchat starts a new instance when I click in a Gnome notification (aka #1262 ).

**Comment by ask-compu** [comment]
when will this be fixed?

**Comment by Arnavion** [comment]
When someone cares enough about it to send a pull request.

**Comment by antivirtel** [comment]
So basically you need this for HexChat too: ![gdbus_xchat](https://cloud.githubusercontent.com/assets/3451979/14275480/b3b785ce-fb17-11e5-900d-f20be0ac8d77.jpg) - am I right? If yes, I can do this, but [for bounty >= 50 USD](https://www.bountysource.com/issues/751287-gdbus-support), the 5 USD is not enough.

**Comment by antivirtel** [comment]
So basically you need this for HexChat too: ![gdbus_xchat](https://cloud.githubusercontent.com/assets/3451979/14275480/b3b785ce-fb17-11e5-900d-f20be0ac8d77.jpg) - am I right? If yes, I can do this, but [for bounty >= 50 USD](https://www.bountysource.com/issues/751287-gdbus-support), the 5 USD is not enough.

**Comment by TingPing** [comment]
@antivirtel No, the only reason people care about this issue is for single instance mode on Windows.

**Comment by ask-compu** [comment]
@Tingling and Linux

**Comment by ask-compu** [comment]
in linux if u click an irc:// link it opens a new instance of hexchat instead of opening the link in the current instance

**Comment by TingPing** [comment]
It works fine in Linux your distro just needs to edit the desktop file, I've done that upstream by default for next release. `hexchat --existing ircs://chat.freenode.net` for example.

**Comment by ask-compu** [comment]
Oh OK, thanks


Sent from my Verizon Wireless 4G LTE smartphone

-------- Original message --------

**Comment by TingPing** [comment]
Finished the easy part porting dbus-client to it: https://github.com/hexchat/hexchat/commit/ec4d3de9d2e4cbc55a0ee1087bc49c14cbe56515

**Comment by amezarashi97** [comment]
Updating HexChat's settings accordingly would be great too.

**Comment by DGMurdockIII** [comment]
Why hasn't this been finshed yet

**Comment by TingPing** [comment]
> Why hasn't this been finshed yet

@DGMurdockIII Rewriting dbus-plugin.c is non-trivial and hell I've never even tested that gdbus works properly on Windows and remember others saying it doesn't; So if you are the individual that cares to do this work, patches welcomed.
